operators/pkg/controller/elasticsearch/driver/default.go

```diff
@@ -364,30 +279,29 @@ func (d *defaultDriver) reconcileNodeSpecs(
 		return results.WithError(err)
 	}
 
+	if !d.expectations.GenerationExpected(actualStatefulSets.ObjectMetas()...) {
+		// Our cache of SatefulSets is out of date compared to previous reconciliation operations.
+		// This will probably lead to conflicting sset updates (which is ok), but also to
+		// conflicting ES calls (set/reset zen1/zen2/allocation excludes, etc.), which may not be ok.
+		log.V(1).Info("StatefulSet cache out-of-date, re-queueing", "namespace", es.Namespace, "es_name", es.Name)
+		return results.WithResult(defaultRequeue)
+	}
+
 	nodeSpecResources, err := nodespec.BuildExpectedResources(es, podSpecBuilder)
 	if err != nil {
 		return results.WithError(err)
 	}
 
-	// TODO: handle zen2 initial master nodes more cleanly
-	//  should be empty once cluster is bootstraped
-	var initialMasters []string
-	// TODO: refactor/move
-	for _, res := range nodeSpecResources {
-		cfg, err := res.Config.Unpack()
-		if err != nil {
-			return results.WithError(err)
-		}
-		if cfg.Node.Master {
-			for i := 0; i < int(*res.StatefulSet.Spec.Replicas); i++ {
-				initialMasters = append(initialMasters, fmt.Sprintf("%s-%d", res.StatefulSet.Name, i))
-			}
-		}
+	// TODO: there is a split brain possibility here if going from 1 to 3 masters or 3 to 7.
```

Comment:
I'm not sure we need to do it in this PR, but it is kind of tempting just to block unsafe scale events. But also since this is just zen1 I wonder if we can just add a disclaimer somewhere

Reply:
Can ignore this, we can discuss it in the overall issue since it's being discussed there and in a few other threads on this PR